src/test/util/setup_common.cpp

```diff
@@ -108,6 +108,8 @@ TestingSetup::TestingSetup(const std::string& chainName) : BasicTestingSetup(cha
     threadGroup.create_thread(std::bind(&CScheduler::serviceQueue, &scheduler));
     GetMainSignals().RegisterBackgroundSignalScheduler(scheduler);
 
+    g_rpc_node->scheduler = &scheduler;
```

Comment:
scheduler is initialized as a member var of the `TestingSetup` struct [here](https://github.com/amitiuttarwar/bitcoin/blob/master/src/test/util/setup_common.h#L88)

only place I've found that use that scheduler is in `denailofservice_tests.cpp` (other than the scheduler tests which instantiate a new scheduler)

my understanding is that adding to `g_rpc_node` here points to the same scheduler, aka just adds another access pattern. so I left the existing initialization & invocation alone. But I'm not sure if thats a bad design pattern.

Reply:
This should go down to where it says `m_node.mempool = &::mempool`.

It then reads `m_node.scheduler = &m_scheduler` or so.

Reply:
fixed by making `unique_ptr`,  so have to keep it up here since the next couple lines start the scheduler thread